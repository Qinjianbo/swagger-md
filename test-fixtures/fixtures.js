import { create as createFixtureLoader } from 'fixture-loader';

const fixture_loader = createFixtureLoader('./test-fixtures/fixtures');

const api = {

  loadSwaggerSpec(filename) {
    return fixture_loader.loadParsedJson('swagger', filename);
  },

};

export default api;
