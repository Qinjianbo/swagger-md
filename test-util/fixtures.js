import { create as createFixtureLoader } from 'fixture-loader';

const fixture_loader = createFixtureLoader('./test-util/fixtures');

const api = {

  loadSwaggerSpec(filename) {
    return fixture_loader.loadParsedJson('swagger', filename);
  },

};

export default api;
