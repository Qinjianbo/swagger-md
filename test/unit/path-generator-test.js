import fixtures from '../../test-util/fixtures';
import generator from '../../src/generators/path-generator';

describe('test/unit/path-generator-test.js', () => {

  describe('generatePath', () => {

    describe('with an API spec having mixed properties; some defined, some omitted', () => {
      let path_api_spec;
      let swagger_api_md_str;

      beforeEach(() => {
        path_api_spec = fixtures.loadSwaggerSpec('path/mixed-properties');
      });

      beforeEach(() => {
        swagger_api_md_str = fixtures.loadSwaggerSpecMarkdown('path/mixed-properties.md');
      });

      it('should render markdown and omit unspecified data, e.g. descriptions, schemas', () => {
        const key = Object.keys(path_api_spec)[0];
        const markdown_str = generator.generatePath(key, path_api_spec[key]);
        markdown_str.should.eql(swagger_api_md_str);
        const markdown_str_with_added_newline = `${markdown_str}\n`;
        markdown_str_with_added_newline.should.eql(swagger_api_md_str);
      });

    });

  });

});
