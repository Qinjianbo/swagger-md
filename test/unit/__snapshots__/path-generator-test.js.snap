exports[`test/unit/path-generator-test.js generatePath with an API spec having mixed properties; some defined, some omitted should render markdown and omit unspecified data, e.g. descriptions, schemas 1`] = `
"### GET /pets/{id}

Returns a user based on a single ID, if the user does not have access to the pet

**Parameters**

| in   | name | type           | required | description        |
|------|------|----------------|----------|--------------------|
| path | id   | integer, int64 | true     | ID of pet to fetch |

#### Response: 200

pet response

**Schema**

- [Pet](#pet)

#### Response: default

unexpected error

**Schema**

- [Error](#error)

### DELETE /pets/{id}

> :warning: **deprecated**

deletes a single pet based on the ID supplied

More info on deleting pets

https://example.com/more-info

**Parameters**

| in   | name | type           | required |
|------|------|----------------|----------|
| path | id   | integer, int64 | true     |

#### Response: 204

pet deleted

**Schema**

N/A

#### Response: default

unexpected error

**Schema**

- [Error](#error)"
`;

exports[`test/unit/path-generator-test.js generatePath with body parameter having a schema should include schema definition in result 1`] = `
"### POST /pets

Creates a new pet

**Parameters**

| in   | name | required | description        |
|------|------|----------|--------------------|
| body | body | true     | Params for new pet |

**Request Body**

- (object)
  - name (string)

#### Response: 201

pet response

**Schema**

- [Pet](#pet)

#### Response: default

unexpected error

**Schema**

- [Error](#error)"
`;
